### PR TITLE
Create narrative.test.js and fill coverage gaps in awards and lineage tests

### PR DESCRIPTION
**Pull request by mikebud8**
## What changed

- **`src/narrative.test.js` (new):** Moves the `buildTickerMessages` describe block out of `gameLogic.test.js`, where it imported directly from `narrative.js` but had no dedicated home. Adds new tests covering every remaining uncovered branch: the complicated-legacy stat message (`wins >= 3 && losses >= 3`), three-combatant win rounds (`l2` branch), bot player/combatant filtering, null-safety fallbacks on `rounds`/`players`/`combatants`, and draw rounds with missing `combatants` property.
- **`src/gameLogic.test.js`:** Removes the `buildTickerMessages` import and describe block (now in `narrative.test.js`). No other changes.
- **`src/awards.test.js`:** Adds tests for `mvp_record: null` fallback in `computeSuperlatives`, missing `combatants` property in `getSeriesCombatantNominees`, missing `rounds`/`players` properties and bot-combatant skip in `computeGameAutoAwards`, draw-round tally path, null rooms arg, `winner.ownerName` fallback, multi-evolution counting, and `standings.length > 0` false branch.
- **`src/lineage.test.js`:** Adds tests for co-parent already in the primary family (seen-set skip), room with no `rounds` property in `buildActiveFormMap` and `buildChainEvolutionStory`, evolution round with no `combatants` property, and merge with `fromNames: null` to exercise the `|| []` and `|| ''` fallbacks on lines 204-206.

## Coverage results

| File | Branch before | Branch after |
|---|---|---|
| `narrative.js` | 76% | 100% |
| `awards.js` | 82% | 99% |
| `lineage.js` | 89% | 99% |

The two remaining uncovered branches (`awards:171`, `lineage:205`) are unreachable defensive guards — the `else if (round.draw)` false-arm inside a loop already filtered for `winner || draw`, and `fromIds || []` behind an `ancestorIdx === -1` guard above it.

## Test notes
- `npx vitest run`: 470 tests, 0 failures (was 429)
- `npm run lint`: 0 errors, 1 pre-existing warning in `scripts/coverage/block-navigation.js`

### CODE REVIEW

File: src/awards.test.js [hunk 1]
```diff
@@ -479,3 +479,264 @@ describe('computeSeasonAutoAwards', () => {
...
+  })
+})
+
+// ─── getSeriesCombatantNominees – edge branches ───────────────────────────────
+
+describe('getSeriesCombatantNominees – edge branches', () => {
+  it('handles room with no combatants property', () => {
+    expect(() => getSeriesCombatantNominees([{ }])).not.toThrow()
+    expect(getSeriesCombatantNominees([{ }])).toEqual([])
+  })
```
[the remark this answers is not in the document]
Comment (mikebud8) [reply src/awards.test.js:499]: Dismissed — the test IS the guard. Adding extra validation inside a test that verifies null-safety behavior would just shadow the real assertion.

File: src/narrative.test.js [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+  })
+
+  it('handles room with no combatants property', () => {
+    const room = { devMode: false, players: [], rounds: [] }
+    expect(() => buildTickerMessages([room])).not.toThrow()
+  })
+
+  it('does not generate stat messages for bot combatants', () => {
+    const bot = makeCombatant({ id: 'c1', name: 'BotFighter', wins: 10, losses: 0, isBot: true })
+    const room = { devMode: false, players: [], combatants: { p1: [bot] }, rounds: [] }
```
[the remark this answers is not in the document]
Comment (mikebud8) [reply src/narrative.test.js:168]: Dismissed — these three tests are intentionally separate so a failure names the exact missing property. Consolidating into one test would produce a single failure message with no indication of which input caused it.
